File: Tests/Model/NetElems/Utilities.fs
module Synapses.Tests.Model.NetElems.Utilities

open FSharpx.Collections
open Synapses.Model
open Synapses.Model.NetElems

let realizedActivFun
        (activationF: Activation.Function)
        : string =
        activationF.name

let realizedNeuron
        (neuron: Neuron.Neuron)
        : string * List<string> =
        (
                realizedActivFun
                       neuron.activationF,
                neuron.weights
                |> LazyList.map (sprintf "%f")
                |> LazyList.toList
        )

let realizedLayer
        (layer: Layer.Layer)
        : List<string * List<string>> =
        layer
        |> LazyList.map realizedNeuron
        |> LazyList.toList

let realizedNetwork
        (network: Network.Network)
        : List<List<string * List<string>>> =
        network
        |> LazyList.map realizedLayer
        |> LazyList.toList

File: implementation/Python/test_dir/model/net_elems/utilities.py
from typing import List

from model.net_elems.activation import Activation
from model.net_elems.neuron import Neuron


def realized_activ_fun(activation_val: Activation)-> str:
    return activation_val.name

def realized_neuron(neuron_val: Neuron)-> (str, List[float]):
    return (
        realized_activ_fun(neuron_val.activation_f),
        neuron_val.weights.to_list()
    )
